### PR TITLE
Adding an injection for hashes

### PR DESCRIPTION
**Pull request by ecwillis**
Why is this pull request necessary:
1. It adds an injection for static files when using file hashes.

Changes proposed in this pull request:
- injection script
- bug fix

### CODE REVIEW

File: lib/urlHashes.js [hunk 1]
```diff
@@ -1,17 +1,19 @@
 'use strict';
-const hash = require('rev-hash');
```
Comment (jgallen23) [inline lib/urlHashes.js]: If we are going to remove this, make sure to do it in package.json too

File: lib/injectHash.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+const glob = require('glob');
+const fs = require('fs');
+
+module.exports = function(fileName, hashedName, conf) {
+  const nameArr = fileName.split('.');
+  const regex = new RegExp(`(\\<\\!--\\s{0,}clientkit:${nameArr.join('\\.')}\\s{0,}--\\>)([\\t\\n\\s\\<a-zA-Z0-9\\"\\/\\.\\>\\=-]*)(\\<\\!--\\s{0,}clientkit:end\\s{0,}-->)`);
+  const fileLink = `<link rel="stylesheet" href="/${conf.prefix}${hashedName}">`;
+  glob(conf.file, (err, files) => {
+    if (err) {
+      console.log('Error finding injection files');
```
Comment (jgallen23) [inline lib/injectHash.js]: should use logr like every thing else

File: lib/injectHash.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    files.forEach((file) => {
+      fs.readFile(file, (fileErr, content) => {
+        let contentString = content.toString();
+        contentString = contentString.replace(regex, `\$1\n${fileLink}\n\$3`);
+        fs.writeFile(file, contentString, (writeErr) => {
```
Comment (jgallen23) [inline lib/injectHash.js]: This writes the file every time. Should we check if it exists first?

File: lib/urlHashes.js (hunk 1 above)
Comment (ecwillis) [inline lib/urlHashes.js]: Done.

File: lib/injectHash.js (hunk 3 above)
Comment (ecwillis) [inline lib/injectHash.js]: Fixed it so it write less. I am overwriting the original file so it uses debounce to most likely write it once.

File: lib/injectHash.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+  glob(conf.files, (err, files) => {
+    if (err) {
+      log('Error finding injection files');
+      return;
+    }
+
+    files.forEach((file) => {
+      let contentString = cache[file];
+      if (!cache[file]) {
```
Comment (jgallen23) [inline lib/injectHash.js]: I'm a little concerned with this.  What if the view html changes?

File: lib/urlHashes.js [hunk 5]
```diff
@@ -1,17 +1,12 @@
...
 let map = {};
 
 module.exports.hash = (inputName, inputContent) => {
-  let hashKey;
-  if (typeof inputContent === 'string') {
-    hashKey = hash(Buffer.from(inputContent));
-  } else {
-    hashKey = hash(inputContent);
-  }
+  const hashKey = crypto.createHash('md5').update(inputContent).digest('hex').slice(0, 10);
```
Comment (jgallen23) [inline lib/urlHashes.js]: so will this work when inputContent is a string or buffer?

File: public/index.html [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
```
Comment (jgallen23) [inline public/index.html]: can we move this into the test directory?

File: lib/injectHash.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    cli: {
+      lineColor: 'yellow'
+    }
+  }
+});
+
+const cache = {};
+
+const writeFile = debounce((file, contentString) => {
+  fs.writeFileSync(file, contentString);
```
Comment (jgallen23) [inline lib/injectHash.js]: now that I think about it, we are already deboucing higher up the funnel, so we might not need this

File: commands/run.js [hunk 8]
```diff
@@ -1,17 +1,62 @@
 'use strict';
+const async = require('async');
+const _ = require('lodash');
```
Comment (jgallen23) [inline commands/run.js]: lodash isn't in package.json

File: commands/run.js [hunk 9]
```diff
@@ -1,17 +1,62 @@
...
   // Tasks
-  mkdirp.sync(config.core.dist);
-  if (config.stylesheets) {
-    Object.keys(config.stylesheets).forEach(style => cssProcessor.runTaskAndWrite(config, process.cwd(), style, config.stylesheets[style]));
-  }
-  if (config.scripts) {
-    Object.keys(config.scripts).forEach(script => jsProcessor(config, process.cwd(), script, config.scripts[script]));
-    lintProcessor(config, process.cwd());
-  }
+  async.auto({
```
Comment (jgallen23) [inline commands/run.js]: will this work for the watchers too? `--env dev`

File: commands/run.js [hunk 10]
```diff
@@ -1,17 +1,62 @@
 'use strict';
+const async = require('async');
+const _ = require('lodash');
 const mkdirp = require('mkdirp');
+const cssProcessor = require('../tasks/css.js');
```
Comment (jgallen23) [inline commands/run.js]: these where inside runAll because they take forever to require

**Comment by jgallen23** [comment]
```
07:21 $ ~/libraries/clientkit/app.js --env dev
  Using local config directory: /home/ubuntu/clients/cuddle-bright/splash/clientkit
  Changed: clientkit/default-colors.yaml
  Changed: clientkit/default-spacing.yaml
  Changed: clientkit/default-vars.yaml
  Changed: clientkit/default.yaml
  Changed: styles.css
  Processed ../../../libraries/clientkit/styles/default.css in 3.064 sec
  Wrote: dist/clientkit.css (64.54 Kb), 
(node:10743) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: cb is not a function
  Processed styles.css in 5.103 sec
  Wrote: dist/styles.css (78.64 Kb), 
(node:10743) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): TypeError: cb is not a function
```

File: commands/run.js (hunk 8 above)
Comment (ecwillis) [inline commands/run.js]: Added

**Comment by ecwillis** [comment]
I added some `noops` to the processors that will take care of the missing `cb` issue.

**Pull request closed without merging**